src/hound_writer.rs

```diff
@@ -0,0 +1,201 @@
+                return Err(ComponentError::HoundError(hound_error));
+            }
+        }
+    }
+}
+
+impl ToString for HoundWriter {
+    /// Converts the HoundWriter to a String, i.e. returns its name
+    fn to_string(&self) -> String {
+        "HoundWriter".to_string()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::component::run_component;
+    use hound::{SampleFormat, WavReader};
+
+    use std::f32::consts::PI;
+    use std::fs::remove_file;
+    use std::{io::BufReader, sync::mpsc::channel};
+
+    const SAMP_RATE: u32 = 44100;
+    const BITS_PER_SAMPLE: u16 = 32;
+    const FRAME_SIZE: i32 = 128;
+
+    const C: f32 = 261.61;
+
+    fn create_sine_wave(frames: i32, note: f32) -> Vec<f32> {
+        (0..frames)
+            .map(|x| (x % FRAME_SIZE) as f32 / FRAME_SIZE as f32)
+            .map(|t| (t * note * 2.0 * PI).sin() * (i16::MAX as f32))
+            .collect()
+    }
+
+    // Write 100 sine wav frames to an output file using a WavWriter, and
+    // read it back properly using a WavReader
+    #[test]
+    fn test_wav_writer_reader() {
+        let file_name = "/Users/Skylar.Gilfeather/CyberGrape/sine.wav";
+
+        let left_samps: Vec<f32> = create_sine_wave(100, C);
+        let right_samps: Vec<f32> = create_sine_wave(100, C);
+
+        // Create WavWriter and write 2 channel sine wav to output file
+        let spec = WavSpec {
+            channels: 2,
+            sample_rate: SAMP_RATE,
+            bits_per_sample: BITS_PER_SAMPLE,
+            sample_format: SampleFormat::Float,
+        };
+
+        let mut writer = WavWriter::create(file_name, spec).unwrap();
+
+        for (&left, &right) in std::iter::zip(&left_samps, &right_samps) {
+            writer.write_sample(left).unwrap();
+            writer.write_sample(right).unwrap();
+        }
+
+        assert!(writer.finalize().is_ok());
+
+        // Create WavReader and read interleaved data back into 2 channel data
+        let mut reader = WavReader::open(file_name).unwrap();
+
+        let all_samps = reader
+            .samples::<f32>()
+            .collect::<Result<Vec<f32>, hound::Error>>()
+            .unwrap();
+
+        let left_samps_out = all_samps
+            .clone()
+            .into_iter()
+            .step_by(2)
+            .collect::<Vec<f32>>();
+        let right_samps_out = all_samps
+            .into_iter()
+            .skip(1)
+            .step_by(2)
+            .collect::<Vec<f32>>();
+
+        assert_eq!(left_samps, left_samps_out);
+        assert_eq!(right_samps, right_samps_out);
+
+        assert!(remove_file(file_name).is_ok());
+    }
+
+    /// Write 100 sine wav frames to an output file using a HoundWriter
+    /// running as a Component thread. Then, read the data back using a
+    /// WavReader and assert that all values are the same.
+    #[test]
+    fn test_hound_writer_read_write_one_frame() {
+        let file_name = "/Users/Skylar.Gilfeather/CyberGrape/sine.wav";
```

Comment:
```suggestion
        let file = NamedTempFile::new().unwrap();
        let file_name = file.path();
```

Reply:
This should make it pass the CI by using a dynamic temporary file instead of a hard-coded path.